.github/workflows/coverage.yml

```diff
@@ -0,0 +1,219 @@
+name: Coverage Report
+
+on:
+    pull_request:
+    push:
+        branches: [ main ]
+
+jobs:
+  coverage:
+    name: Generate Coverage Report
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read          # To checkout the code
+      pull-requests: write    # To comment on PRs
+      issues: write          # To comment on issues (PRs are issues)
+      actions: read          # To read workflow artifacts
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # Fetch full history for diff coverage analysis
+          fetch-depth: 0
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+        with:
+          enable-cache: true
+
+      - name: Set up Python 3.11
+        run: uv python install 3.11
```

Comment:
Why not 3.13? That's what we use for building apps, etc. Only core lib is back compatible to older versions.